### DISCUSSION TITLE
In case of using a plugin for calibration, missed data are sent as raw-data to aaps

### DISCUSSION THREAD

**Discussion by aptash** [NightscoutFoundation/xDrip#3174] [Bug reports]
Hello!

We use Sibionics CGM. For this monitoring we use 670G/EverSense source and for calibration we have to use a plugin.
xDrip sends the data to aaps.

The problem:
in case of connection problems, after the connection is restored only the last value is sent calibrated to aaps. All other missed data are sent as raw-values to aaps.

You can see the problem on this screenshot.
Aaps 3.2 sent data to Nightscout.

![image](https://github.com/NightscoutFoundation/xDrip/assets/5793551/118cf438-bb6c-41d2-a772-e5cf78a94cea)

It seems like a bug to me or can we do something about it?

**Comment by aptash** [comment]
To emphasise the seriousness of the bug I decided to attach one more screenshot:

![image](https://github.com/NightscoutFoundation/xDrip/assets/5793551/91030b57-4fe1-4581-b38d-4dc4e6c863d6)

As you can see after missed data are got the difference between the last BG and previous can be huge.
Of course aaps takes that fact into the consideration and starts its actions immediately. It could be an increased basal rate and/or SMB. Which can lead to dangerous consequences.

**Reply by Navid200** [reply]
Sibionics CGM does not offer a calibration option?

**Reply by aptash** [reply]
No. It doesn't.
The original app SiJoy can't upload data to xDrip.
The patched version of SiJoy is called SugarJoy+ uploads data to xDrip as "EverSense"-source.
xDrip doesn't offer calibrations for this kind of source. So we have to use a calibration-plugin.

If your question was about Sibionics sensors - they are factory calibrated as well as Freestyle Libre sensors.
But in my case (and many other people), Sibionics sensors show the data about 2 mmol below the finger prick glucose measurement. That is why I want to calibrate the data.

**Reply by jamorham** [reply]
The values that you are saying are misreported, is this back filled data being sent by the 3rd party app?

**Reply by aptash** [reply]
Yes. These data are sent by the patched app SugarJoy+.
They are displayed ok (calibrated) in xDrip but are sent as raw to aaps.

If I use xDrip-follower (as a clone on the same phone) then all the data are calibrated in xDrip-follower. But xDrip-follower doesn't sent backfilled data altogether to NS or aaps.

**Reply by jamorham** [reply]
So is the data you show in the screenshots on nightscout, does this get sent to nightscout by AAPS or sent to nightscout by xDrip?

I think the most likely is that the data is going through a display glucose filter which probably only works on the most recent value.

**Reply by aptash** [reply]
As I said in the original message:
> Aaps 3.2 sent data to Nightscout.

My schema is:
SugarJoy+ -> xDrip -> AAPS -> NS

**Reply by Inkalika** [reply]
I think the problem is that the active plugin only performs the function of displaying calibrated sugar on the graph xDrip, but when transferred to Nigtscout, these values are transferred uncalibrated. I think the solution is to expand the functionality of active plugins not only by displaying calibrated sugar, but also by transferring these values to Nightscout. And also to correct situations where, due to calibrations through the active plugin, readings on followers (xDrip follower) begin to be transmitted in the form of “???”, but after repeated initial calibration on the master, this situation is corrected

My schema is:
SugarJoy+ -> xDrip -> NS 
SugarJoy+ -> xDrip -> AAPS

**Comment by Navid200** [comment]
There was a time calibration was necessary.  Back then, sensor had no means of compensating for the impact of the immune system on the sensor filament.
xDrip offered to calibrate as it was a required part of using a CGM.

The CGM technology has evolved.
The new devices don't need to be calibrated.
Unfortunately, there are individuals who fall outside the statistical spread of the new mechanism.  Hence, the calibration option still exists.
An example is Dexcom G7.

Unfortunately, there are individuals who refuse to accept the evolution and insist to calibrate regardless.  The problem is that the blood glucose meter has its own inaccuracies.  It was never expected that it would be extremely accurate.  But, for the first generation CGMs, the meter was the only option and it was good enough to be used as the point of reference considering how bad the drift due to the immune system was.

But, now, individuals who insist to calibrate regardless, may not realize that their blood glucose meter is not perfect and their CGM is not meant to match it.

We (xDrip) have offered flexibility to the user.  But, if we overdo that, we may actually do a disservice.

Please find out how a company that has released a product that is supposed to have passed medical tests and verifications claims that there is no need for calibration.
How do the doctors respond to that claim if it is an incorrect claim?
How did they get approval to release the product?

**Reply by Inkalika** [reply]
In the first few days, the sensor underestimates by 2-3 mmol (Libra2, for example, has only a few hours), then it constantly underestimates by 1-1.5 mmol (Libra2 differs by no more than 0.5 mmol). And here it's not that there are deviations with the glucose meter, namely systematic deviations and values go beyond the error zone. The manufacturer assures that their sensors are accurate and do not need calibration, but many users assure the opposite. And everyone's sensor underestimates, but everyone has their own mmol value.

**Reply by yiyoMX** [reply]
For what I know sensor sends a raw value to the app, and the app processes the value and 'calibrates it', and compensates for temperature and other factors. So, if the value comes from the app it should be calibrated. I've found Sibionics sensor to be pretty accurate.

**Reply by aptash** [reply]
In our case Sibionics constantly shows 2-3 mmol below a glucose meter. Currently we are using 3rd sensor.
